examples/bri-3/src/bri/transactions/agents/transactions.agent.ts

```diff
@@ -299,4 +291,86 @@ export class TransactionAgent {
 
     return name;
   }
+
+  private async prepareCircuitInputs(
+    tx: Transaction,
+    circuitInputsTranslationSchema: string,
+  ): Promise<object> {
+    const payloadAsCircuitInputs = await this.preparePayloadAsCircuitInputs(
+      tx.payload,
+      circuitInputsTranslationSchema,
+    );
+
+    return Object.assign(
+      payloadAsCircuitInputs,
+      await computeEddsaSigPublicInputs(tx),
+    );
+  }
+
+  private async preparePayloadAsCircuitInputs(
+    txPayload: string,
+    workstepTranslationSchema: string,
+  ): Promise<object> {
+    const mapping: CircuitInputsMapping = JSON.parse(workstepTranslationSchema);
+
+    if (!mapping) {
+      throw new Error(`Broken mapping`);
+    }
+
+    const parsedInputs =
+      this.circuitInputsParserService.applyMappingToJSONPayload(
+        txPayload,
+        mapping,
+      );
+    if (!parsedInputs) {
+      throw new Error(`Failed to parse inputs`);
+    }
+
+    return parsedInputs;
+  }
+
+  private constructTxHash(
+    merkelizedPayload: MerkleTree,
+    witness: Witness,
+  ): string {
+    const hashFn = this.merkleTreeService.createHashFunction(
+      `${process.env.MERKLE_TREE_HASH_ALGH}`,
+    );
+
+    const merkelizedInvoiceRoot = merkelizedPayload.getRoot().toString('hex');
+    const witnessHash = hashFn(JSON.stringify(witness)).toString('hex');
+
+    return hashFn(`${merkelizedInvoiceRoot}${witnessHash}`).toString('hex');
+  }
 }
+// TODO: Example input preparation for other workstep circuits from the example use-case, to be used
```

Comment:
@ognjenkurtic is that gonna be an issue you create?

Reply:
https://github.com/ethereum-oasis-op/baseline/issues/804